XenAdmin/Program.cs

```diff
@@ -821,7 +821,7 @@ public static void Invoke(Control c, MethodInvoker f)
             {
                 if (!IsExiting(c))
                 {
-                    if (c.InvokeRequired)
+                    if (c != null && c.InvokeRequired)

```

Comment:
I think it would be better to keep this as it was and just have a 

if (c == null)
   return;

so we don't get to the else to call f() when c is "accidentally" null at this point